### ISSUE TITLE
Invalid Authentication - Error 2406

### ISSUE THREAD

**Issue by sthorley** [tuya/tuya-home-assistant#40]
Hi, Following the announcement on the HomeAssistant Reddit I signed up for Tuya IOT and installed Tuya V2 version 1.3 using HACS on July 1st 2021. Once I signed up for Tuya IOT I signed up for the Free api, created my first project, linked my "TuyaSmart" app account, and granted the API permissions. I can see my Tuya devices have successfully linked in the Tuya IOT project page.  I have never had a Tuya IOT project before 1st July 21 so there is no way I was using the older authentication method.

Despite this I am getting an error 2406 in my logs. I have tried reinstalling the Custom Component, creating a new Tuya IOT project and resetting my Tuya password. None of these things have worked. I am using an Australian account and have specified the "Europe" availability region during the setup.

Logs:
`2021-07-01 16:35:40 INFO (MainThread) [custom_components.tuya_v2.config_flow] TuyaConfigFlow.async_step_user start, is_import= None
2021-07-01 16:36:06 INFO (MainThread) [custom_components.tuya_v2.config_flow] TuyaConfigFlow.async_step_user start, is_import= {'password': 'PASSWORD', 'endpoint': 'https://openapi.tuyaeu.com', 'access_id': 'p4qXXXXXXXXXXdnt', 'access_secret': 'dffacdd97c0XXXXXXXXXX15b07c86', 'tuya_app_type': 'tuyaSmart', 'country_code': '61', 'username': 'xxx@sxxxy.xom'}
2021-07-01 16:36:06 INFO (SyncWorker_4) [custom_components.tuya_v2.config_flow] TuyaConfigFlow._try_login start, user_input: {'password': 'PASSWORD', 'endpoint': 'https://openapi.tuyaeu.com', 'access_id': 'p4qXXXXXXXXXXdnt', 'access_secret': 'dffacdd97c0XXXXXXXXXX15b07c86', 'tuya_app_type': 'tuyaSmart', 'country_code': '61', 'username': 'xxx@sxxxy.xom', 'tuya_project_type': 0}
2021-07-01 16:36:08 INFO (SyncWorker_4) [custom_components.tuya_v2.config_flow] TuyaConfigFlow._try_login finish, response:, {'code': 2406, 'msg': 'skill id invalid', 'success': False, 't': 16251XXXXX19}`

**Comment by robertbigk** [comment]
Wrong contry code use like PL , EN , GB check  Tuya code list

**Comment by sthorley** [comment]
> Wrong contry code use like PL , EN , GB check Tuya code list I also do that same and cant make account work:)

No. You are incorrect. All the configuration examples, screenshots, developer comments, and every piece of documentation I have read shows this as the IDD Telephone code of the country. Eg: 86 for China, 1 for USA, 61 for Australia.

For completeness I have tested “AU” and the 2406 issue remains. 

Please do not comment on open issues unless you can substantiate your claims with evidence or at least have it working yourself with that config…

**Comment by robertbigk** [comment]
ok just try to help. Today I add an account I create 3 accounts and cant connect I try all time code digits and got errors, in-home assistant. I use code letter in my case PL and account connected to the home assistant.

**Comment by osk2** [comment]
Try to unlink project in IoT platform and link again, also reset password in app. That works for me.

**Comment by stevekuchta** [comment]
Resetting the Tuya app password (not the developer password) seems to have fixed this issue for me. I did this from within the app. I had already unlinked and relinked the account from within the Tuya developer console, but I don't think that had an impact. I did not unlink and relink after I reset the Tuya app password. (Also of note is that I had both my Tuya app and developer passwords the same before doing this and now they are different, but I don't know if that has any impact.)
